From: Peter Samuelson <peter@p12n.org>
Date: Thu, 13 Jul 2006 09:40:45 -0500
Subject: berkeley-db.m4: Avoid bdb version detection; just link to libaprutil instead

In #378105 I reported that 'apu-config --libs' prints a list of
dependent libraries, where it should print nothing at all, since you
should not explicitly link to dependent libraries in Debian.  Tollef
said he'd fix it.

However, BDB version detection relies on this behavior, as it runs
'apu-config --libs', pulls out the -ldb* word, and compiles a test
program with this.  By adding 'apu-config --link-ld', which will be
-laprutil-1, that will pull in libdb4.x regardless of whether
'apu-config --libs' prints it explicitly.

I don't think this can be sent upstream, as it won't work when
compiling aprutil in-tree.  But I'm not certain.

@DPATCH@
Index: subversion-1.7.0/build/ac-macros/berkeley-db.m4
===================================================================
--- subversion-1.7.0.orig/build/ac-macros/berkeley-db.m4	2011-10-20 20:45:56.000000000 +0200
+++ subversion-1.7.0/build/ac-macros/berkeley-db.m4	2011-10-20 21:55:55.000000000 +0200
@@ -183,7 +183,7 @@
     SVN_DB_LIBS=["${SVN_DB_LIBS-`$apu_config --libs | $SED -e 's/.*\(-ldb[^[:space:]]*\).*/\1/' | $EGREP -- '-ldb[^[:space:]]*'`}"]
 
     CPPFLAGS="$SVN_DB_INCLUDES $SVN_APRUTIL_INCLUDES $CPPFLAGS" 
-    LIBS="`$apu_config --ldflags` $SVN_DB_LIBS $LIBS"
+    LIBS="`$apu_config --ldflags --link-ld` $SVN_DB_LIBS $LIBS"
 
     if test -n "$svn_berkeley_db_header"; then
       SVN_DB_HEADER="#include <$svn_berkeley_db_header>"
